### PR TITLE
Implement sending message from the linux layer of the android tv-app::platform-app to the tv-app::content-app

### PR DESCRIPTION
**Pull request by amitnj**
#### Problem
Messages coming to the tv-app::platform-app were not being forwarded to the content-app. They worked only for the endpoint 1 and not for the content app endpoints.

#### Change overview
Created the required JNI channels to get the message to the java layer and then on to the content app using the intent mechanism introduced earlier. Also modified the discovery mechanism to add the content apps discovered to the ContentAppPlatform of the core matter SDK which would enable these apps having their own endpoints which can be targeted.

#### Testing
Manually tested using the UI affordances built in the tv-app::platform-app. Introduced a mechanism to invoke the full message communication from the command handlers to the content app using that UI.

### CODE REVIEW

File: examples/tv-app/android/include/content-launcher/AppContentLauncherManager.cpp [hunk 1]
```diff
@@ -53,8 +57,11 @@ void AppContentLauncherManager::HandleLaunchUrl(CommandResponseHelper<LaunchResp
     string displayStringString(displayString.data(), displayString.size());
 
     // TODO: Insert code here
+
+    const char * resStr = mCommandDelegate.sendCommand(mEndpointId, contentUrlString);
```
Comment (chrisdecenzo) [inline examples/tv-app/android/include/content-launcher/AppContentLauncherManager.cpp:61]: what happens here when the commandDelegate is null (eg. the case with the 4 dummy AppImpls in AppImpl.h)?

File: examples/tv-app/android/java/AppImpl.h [hunk 2]
```diff
@@ -127,11 +142,11 @@ class DLL_EXPORT ContentAppFactoryImpl : public ContentAppFactory
     CHIP_ERROR ConvertToPlatformCatalogVendorApp(const CatalogVendorApp & sourceApp, CatalogVendorApp * destinationApp) override;
 
 protected:
-    ContentAppImpl mContentApps[APP_LIBRARY_SIZE] = {
-        ContentAppImpl("Vendor1", 1, "exampleid", 11, "Version1", "34567890"),
-        ContentAppImpl("Vendor2", 65521, "exampleString", 32768, "Version2", "20202021"),
-        ContentAppImpl("Vendor3", 9050, "App3", 22, "Version3", "20202021"),
-        ContentAppImpl("TestSuiteVendor", 1111, "applicationId", 22, "v2", "20202021")
+    std::vector<ContentAppImpl> mContentApps{
+        ContentAppImpl("Vendor1", 1, "exampleid", 11, "Version1", "34567890", nullptr),
```
Comment (chrisdecenzo) [inline examples/tv-app/android/java/AppImpl.h:146]: don't these nullptr arguments cause problems in the ContentAppEndpointManagerDelegate constructor?

**Review by chrisdecenzo — APPROVED** [review]

File: examples/tv-app/android/java/TVApp-JNI.cpp [hunk 3]
```diff
@@ -175,3 +175,27 @@ JNI_METHOD(jboolean, setCurrentLevel)(JNIEnv *, jobject, jint endpoint, jboolean
...
 }
+
+JNI_METHOD(jint, addContentApp)
+(JNIEnv *, jobject, jstring vendorName, jint vendorId, jstring appName, jint productId, jstring appVersion, jobject manager)
+{
+    JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
+
+    const char * nVendorName = env->GetStringUTFChars(vendorName, 0);
+    const char * nAppName    = env->GetStringUTFChars(appName, 0);
+    const char * nAppVersion = env->GetStringUTFChars(appVersion, 0);
```
Comment (nicelyjust) [inline examples/tv-app/android/java/TVApp-JNI.cpp:186]: maybe you can use JniUtfString replace this

Comment (amitnj) [reply examples/tv-app/android/java/TVApp-JNI.cpp]: @nicelyjust Thanks. I have made the change. Can you please review again and approve if it looks ok.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**